### PR TITLE
[IngestionClient] Update ingestion client to net6.0 and add action to create release binaries

### PR DESCRIPTION
**Pull request by HenryvanderVegte**
## Purpose
* Moving Ingestion client from net core 3 to net 6
* Adding a github action to create release binaries for the Ingestion Client

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

**Review by PanosPeriorellis — DISMISSED** [review]

**Review by rarera — DISMISSED** [review]

File: .github/workflows/ingestion_client.yaml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      - uses: actions/checkout@v2
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '6.0.x'
+
+      - name: Get Release Versions
+        id: get_release_versions
+        run: |
+          version=$(cat ./samples/ingestion/ingestion-client/Setup/ArmTemplateBatch.json | jq '.variables.Version' | tr -d '"')
```
Comment (mahilleb-msft) [inline .github/workflows/ingestion_client.yaml:21]: >jq '.variables.Version' | tr -d '"' [](http://example.com/codeflow?start=91&length=35)

jq -r .variables.version
allows you to drop the tr
[also below]

Comment (mahilleb-msft) [inline .github/workflows/ingestion_client.yaml:21]: this probably runs under Bash?
consider adding

```
set -u -e -o pipefail
```

at the beginning

File: .github/workflows/ingestion_client.yaml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      - name: Clean Solution
+        run: dotnet clean ./samples/ingestion/ingestion-client/BatchIngestionClient.sln --configuration Release && dotnet nuget locals all --clear
+
+      - name: Build BatchIngestionClient
+        run: dotnet build  samples/ingestion/ingestion-client --configuration Release
+
+      - name: Publish StartTranscriptionByTimer function
+        run: dotnet publish -c Release -f net6.0 --no-self-contained --output ./StartTranscriptionByTimer samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionByTimer.csproj
+      - name: Zip StartTranscriptionByTimer
+        run: cd StartTranscriptionByTimer; zip -r ../StartTranscriptionByTimer.zip *
```
[the remark this answers is not in the document]
Comment (HenryvanderVegte) [reply .github/workflows/ingestion_client.yaml]: fixed!

File: samples/ingestion/ingestion-client/.editorconfig [hunk 3]
```diff
@@ -0,0 +1,1362 @@
...
+# Ingestion Client code rules
+# Description: Code analysis rules for Ingestion Client code
+
+# Remove the line below if you want to inherit .editorconfig settings from higher directories
+root = true
+
+# All Files
+
+[*]
+charset = utf-8-bom
```
Comment (mahilleb-msft) [inline samples/ingestion/ingestion-client/.editorconfig]: would you want to change that to utf-8?

Comment (mahilleb-msft) [reply samples/ingestion/ingestion-client/.editorconfig]: (so editorconfig and ci/check-git-head.sh do not fight each other)

File: samples/ingestion/ingestion-client/.editorconfig [hunk 4]
```diff
@@ -0,0 +1,1362 @@
...
+# Wrapping preferences
+csharp_preserve_single_line_blocks = true
+csharp_preserve_single_line_statements = true
+
+# Coding Conventions
+
+dotnet_sort_system_directives_first = true
+csharp_using_directive_placement = inside_namespace:warning
+dotnet_separate_import_directive_groups = true
+file_header_template = <copyright file="{fileName}" company="Microsoft Corporation">\nCopyright (c) Microsoft Corporation. All rights reserved.\n</copyright>
```
Comment (mahilleb-msft) [inline samples/ingestion/ingestion-client/.editorconfig:1244]: default header for cs here should be

```
// <copyright file="DatabaseConnector.cs" company="Microsoft Corporation">
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
// </copyright>
```

?

**Review by mahilleb-msft — DISMISSED** [review]
:shipit:

File: .github/workflows/ingestion_client.yaml [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        uses: actions/github-script@v3
+        with:
+          script: |
+            core.setFailed('Invalid version specified or mismatching versions in batch and realtime.')
+
+      - name: Clean Solution
+        run: dotnet clean ./samples/ingestion/ingestion-client/BatchIngestionClient.sln --configuration Release && dotnet nuget locals all --clear
+
+      - name: Build BatchIngestionClient
+        run: dotnet build  samples/ingestion/ingestion-client --configuration Release
```
Comment (mahilleb-msft) [inline .github/workflows/ingestion_client.yaml:39]: >ld  sa [](http://example.com/codeflow?start=23&length=6)

nit - duplicate space

File: .github/workflows/ingestion_client.yaml [hunk 6]
```diff
@@ -0,0 +1,125 @@
+name: Build Ingestion Client Binaries
+
+on:
+  push:
+    branches: ['main', 'master']
+    paths: ['samples/ingestion/ingestion-client/**', '.github/workflows/ingestion_client.yaml']
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (BrianMouncer) [inline .github/workflows/ingestion_client.yaml:10]: We are supposed to be using 1ES hosted pool, running the secure version of the public agents image.

File: .github/workflows/ingestion_client.yaml (hunk 1 above)
Comment (HenryvanderVegte) [reply .github/workflows/ingestion_client.yaml]: Fixed!

File: samples/ingestion/ingestion-client/.editorconfig (hunk 3 above)
Comment (HenryvanderVegte) [reply samples/ingestion/ingestion-client/.editorconfig]: Fixed!

File: samples/ingestion/ingestion-client/.editorconfig (hunk 4 above)
Comment (HenryvanderVegte) [reply samples/ingestion/ingestion-client/.editorconfig]: Fixed!

File: .github/workflows/ingestion_client.yaml (hunk 5 above)
Comment (HenryvanderVegte) [reply .github/workflows/ingestion_client.yaml]: Fixed!

**Review by mahilleb-msft — APPROVED** [review]
:shipit:

**Pull request merged**